lib/extras/dec/pnm.cc

```diff
@@ -231,7 +231,8 @@ class Parser {
     if (max_val == 0 || max_val >= 65536) {
       return JXL_FAILURE("PAM: bad MAXVAL");
     }
-    header->bits_per_sample = CeilLog2Nonzero(max_val);
+    // e.g When `max_val` is 4096, we want 13 bits:
```

Comment:
Maybe we should return error if maxval is not of the form 2^n - 1. I'm not sure if it is obligatory to do that in pnm, but I know that's what basically all applications will expect, including this one.

Reply:
`CeilLog2Nonzero(1u)` is legal

Reply:
Ok I added the missing `JXL_RETURN_IF_ERROR` stuff